### PR TITLE
[Enhancement] Support publish version for lake table

### PR DESCRIPTION
**Pull request by sduzh**
This implementation is not performant

## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/transaction/DatabaseTransactionMgr.java [hunk 1]
```diff
@@ -1088,44 +1091,6 @@ public int getTransactionNum() {
         return idToRunningTransactionState.size() + finalStatusTransactionStateDeque.size();
     }
 
-    public TransactionState getTransactionStateByCallbackIdAndStatus(long callbackId, Set<TransactionStatus> status) {
```
Comment (sduzh) [inline fe/fe-core/src/main/java/com/starrocks/transaction/DatabaseTransactionMgr.java:1091]: unused method

File: fe/fe-core/src/main/java/com/starrocks/transaction/DatabaseTransactionMgr.java [hunk 2]
```diff
@@ -1088,44 +1091,6 @@ public int getTransactionNum() {
...
-                    return txn;
-                }
-            }
-        } finally {
-            readUnlock();
-        }
-        return null;
-    }
-
-    public TransactionState getTransactionStateByCallbackId(long callbackId) {
```
Comment (sduzh) [inline fe/fe-core/src/main/java/com/starrocks/transaction/DatabaseTransactionMgr.java:1110]: unused method

File: fe/fe-core/src/main/java/com/starrocks/transaction/GlobalTransactionMgr.java [hunk 3]
```diff
@@ -605,27 +605,6 @@ public long loadTransactionState(DataInputStream dis, long checksum) throws IOEx
         return checksum;
     }
 
-    public TransactionState getTransactionStateByCallbackIdAndStatus(long dbId, long callbackId,
```
Comment (sduzh) [inline fe/fe-core/src/main/java/com/starrocks/transaction/GlobalTransactionMgr.java:608]: unused method

File: fe/fe-core/src/main/java/com/starrocks/transaction/OlapTableTxnStateListener.java [hunk 4]
```diff
@@ -198,93 +195,4 @@ public void postWriteCommitLog(TransactionState txnState) {
...
-                        }
-                    }
-                }
-            }
-            partition.setNextVersion(partition.getNextVersion() + 1);
-        }
-    }
-
-    @Override
-    public void applyVisibleLog(TransactionState txnState, TableCommitInfo commitInfo) {
```
Comment (sduzh) [inline fe/fe-core/src/main/java/com/starrocks/transaction/OlapTableTxnStateListener.java:225]: Moved to OlapTableTxnLogApplier.applyVisibleLog() without any change.

File: fe/fe-core/src/main/java/com/starrocks/transaction/OlapTableTxnLogApplier.java [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+public class OlapTableTxnLogApplier implements TransactionLogApplier {
+    private final OlapTable table;
+
+    public OlapTableTxnLogApplier(OlapTable table) {
+        this.table = table;
+    }
+
+    @Override
+    public void applyCommitLog(TransactionState txnState, TableCommitInfo commitInfo) {
```
Comment (sduzh) [inline fe/fe-core/src/main/java/com/starrocks/transaction/OlapTableTxnLogApplier.java:25]: Moved from the old method `OlapTableStateMachine.applyCommitLog` without any change.

File: fe/fe-core/src/main/java/com/starrocks/transaction/OlapTableTxnLogApplier.java [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+                        }
+                    }
+                }
+            }
+            partition.setNextVersion(partition.getNextVersion() + 1);
+        }
+    }
+
+    @Override
+    public void applyVisibleLog(TransactionState txnState, TableCommitInfo commitInfo) {
```
Comment (sduzh) [inline fe/fe-core/src/main/java/com/starrocks/transaction/OlapTableTxnLogApplier.java:47]: Moved from the old method `OlapTableStateMachine.applyVisibleLog` without any change.

**Review by wyb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 94 / 274 (34.31%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/transaction/LakeTableTxnLogApplier.java|0|16|00.00%|
|:large_blue_circle:|com/starrocks/transaction/LakeTableTxnStateListener.java|0|47|00.00%|
|:large_blue_circle:|com/starrocks/transaction/PublishVersionDaemon.java|4|97|04.12%|
|:large_blue_circle:|com/starrocks/rpc/BackendServiceProxy.java|6|19|31.58%|
|:large_blue_circle:|com/starrocks/transaction/TransactionStateListenerFactory.java|4|6|66.67%|
|:large_blue_circle:|com/starrocks/transaction/TransactionLogApplierFactory.java|2|3|66.67%|
|:large_blue_circle:|com/starrocks/transaction/OlapTableTxnLogApplier.java|57|65|87.69%|
|:large_blue_circle:|com/starrocks/transaction/OlapTableTxnStateListener.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/planner/OlapTableSink.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/transaction/DatabaseTransactionMgr.java|17|17|100.00%|
|:large_blue_circle:|com/starrocks/transaction/TransactionState.java|1|1|100.00%|

**Review by wuleistarrocks — APPROVED** [review]

**Comment by sduzh** [comment]
run starrocks_be_unittest

**Review by decster — APPROVED** [review]

**Pull request merged**